src/rt/minfo.d

```diff
@@ -505,12 +513,11 @@ struct ModuleGroup
         }
 
         // finally, do the sorting for both shared and tls ctors.
-        try
-        {
-            _ctors = doSort(MIctor | MIdtor);
-            _tlsctors = doSort(MItlsctor | MItlsdtor);
-        }
-        catch(DeprecatedCycleException)
+        bool cycle = false;
+        _ctors = doSort(MIctor | MIdtor, cycle);
```

Comment:
You need to do this a weird way. The old algorithm sets `_ctors` and `_tlsctors` internally, so the assignment cannot happen. The exception prevented the assignment. You need to do something like this:

```D
auto ctorTemp = doSort(MIctor | MIdtor, cycle);
if(!cycle)
{
    _ctors = ctorTemp;
    ctorTemp = doSort(MItlsctor | MItlsdtor, cycle);
    if(!cycle)
        _tlsctors = ctorTemp;
}
```